sky/clouds/vast.py

```diff
@@ -0,0 +1,296 @@
+            'image_id': image_id,
+        }
+
+    def _get_feasible_launchable_resources(
+        self, resources: 'resources_lib.Resources'
+    ) -> 'resources_utils.FeasibleResources':
+        """Returns a list of feasible resources for the given resources."""
+        if resources.use_spot:
+            # TODO: Add hints to all return values in this method to help
+            #  users understand why the resources are not launchable.
+            return resources_utils.FeasibleResources([], [], None)
+        if resources.instance_type is not None:
+            assert resources.is_launchable(), resources
+            resources = resources.copy(accelerators=None)
+            return resources_utils.FeasibleResources([resources], [], None)
+
+        def _make(instance_list):
+            resource_list = []
+            for instance_type in instance_list:
+                r = resources.copy(
+                    cloud=Vast(),
+                    instance_type=instance_type,
+                    accelerators=None,
+                    cpus=None,
+                )
+                resource_list.append(r)
+            return resource_list
+
+        # Currently, handle a filter on accelerators only.
+        accelerators = resources.accelerators
+        if accelerators is None:
+            # Return a default instance type
+            default_instance_type = Vast.get_default_instance_type(
+                cpus=resources.cpus,
+                memory=resources.memory,
+                disk_tier=resources.disk_tier)
+            if default_instance_type is None:
+                # TODO: Add hints to all return values in this method to help
+                #  users understand why the resources are not launchable.
+                return resources_utils.FeasibleResources([], [], None)
+            else:
+                return resources_utils.FeasibleResources(
+                    _make([default_instance_type]), [], None)
+
+        assert len(accelerators) == 1, resources
+        acc, acc_count = list(accelerators.items())[0]
+        (instance_list, fuzzy_candidate_list
+        ) = service_catalog.get_instance_type_for_accelerator(
+            acc,
+            acc_count,
+            use_spot=resources.use_spot,
+            cpus=resources.cpus,
+            region=resources.region,
+            zone=resources.zone,
+            clouds='vast')
+        if instance_list is None:
+            return resources_utils.FeasibleResources([], fuzzy_candidate_list,
+                                                     None)
+        return resources_utils.FeasibleResources(_make(instance_list),
+                                                 fuzzy_candidate_list, None)
+
+    @classmethod
+    def check_credentials(cls) -> Tuple[bool, Optional[str]]:
+        """ Verify that the user has valid credentials for Vast. """
+        try:
+            import vastai_sdk as _vast  # pylint: disable=import-outside-toplevel
```

Comment:
Should we use adaptors?

Reply:
I'm agnostic as to the aesthetics of the invocation of our library here. As it stands now it works so I'm going to resolve this

Reply:
I mean I was just following the pattern that is used in 

aws:
https://github.com/skypilot-org/skypilot/blob/02fb8bd242b65fe0e7165c8ee900ae6fe0df2150/sky/clouds/aws.py#L558

runpod:
https://github.com/skypilot-org/skypilot/blob/02fb8bd242b65fe0e7165c8ee900ae6fe0df2150/sky/clouds/runpod.py#L247

vsphere:
https://github.com/skypilot-org/skypilot/blob/02fb8bd242b65fe0e7165c8ee900ae6fe0df2150/sky/clouds/vsphere.py#L262

gcp:
https://github.com/skypilot-org/skypilot/blob/02fb8bd242b65fe0e7165c8ee900ae6fe0df2150/sky/clouds/gcp.py#L710

cudo:
https://github.com/skypilot-org/skypilot/blob/02fb8bd242b65fe0e7165c8ee900ae6fe0df2150/sky/clouds/cudo.py#L275

I was thinking this approach was also acceptable for Vast. Am I wrong?

Reply:
I see.. This is surprising. cc @Michaelvll  - any reason not using adaptor here?